src/garage/sampler/__init__.py

```diff
@@ -15,21 +15,12 @@
 from garage.sampler.vec_env_executor import VecEnvExecutor
 from garage.sampler.worker import DefaultWorker, Worker
 from garage.sampler.worker_factory import WorkerFactory
+from garage.tf.samplers.task_embedding_worker import TaskEmbeddingWorker
 
 __all__ = [
-    'BatchSampler',
-    'Sampler',
-    'ISSampler',
-    'singleton_pool',
-    'LocalSampler',
-    'RaySampler',
-    'SamplerWorker',
-    'MultiprocessingSampler',
-    'ParallelVecEnvExecutor',
-    'VecEnvExecutor',
-    'OffPolicyVectorizedSampler',
-    'OnPolicyVectorizedSampler',
-    'WorkerFactory',
-    'Worker',
-    'DefaultWorker',
+    'BatchSampler', 'Sampler', 'ISSampler', 'singleton_pool', 'LocalSampler',
+    'RaySampler', 'SamplerWorker', 'MultiprocessingSampler',
+    'ParallelVecEnvExecutor', 'VecEnvExecutor', 'OffPolicyVectorizedSampler',
+    'OnPolicyVectorizedSampler', 'WorkerFactory', 'Worker', 'DefaultWorker',
+    'TaskEmbeddingWorker'
```

Comment:
please implement functionality-specific workers within the smallest package context which uses them. in this case, it looks like you should implement this worker inside `garage/tf/algos/te_ppo.py` or wherever you are putting the task embeddings implementation.